### PR TITLE
Transpose Convolution Valid/Full Padding

### PR DESCRIPTION
**Pull request by yg320**
Tensporpack implementation for Transpose Convolution is used when TF-version > 1.12.
In this implementation, output shape is calculated without taking padding into consideration.

I added support for valid/full padded Transpose Convolution in this pull request (which is consistent with tf.layers implementation).

### CODE REVIEW

**Comment by ppwwyyxx** [comment]
Thanks for catching this! I just added a unittest in f7a79d48cb00e9bda8d5f6928348d0a90788edc2 for such errors. Could you add "valid/full" to the test as well and make sure it passes?

**Comment by yg320** [comment]
Done.
Also, I decided to remove support for 'full' padding since neither tf.layers.Conv2DTranspose nor tf.nn.conv2d_transpose seem to support that.

**Comment by ppwwyyxx** [comment]
It looks like the commit "Removed NCHW test from TestConv2DTranspose" actually also removes the tests for valid padding.

**Comment by yg320** [comment]
Sorry for that, I added it back
